lib/compress.js

```diff
@@ -950,9 +950,11 @@ merge(Compressor.prototype, {
                         scope = save_scope;
                         return true;
                     }
-                    if (node instanceof AST_SymbolRef || node instanceof AST_PropAccess) {
+                    if (node instanceof AST_PropAccess
+                        || node instanceof AST_SymbolRef
+                        || node instanceof AST_This) {
```

Comment:
@kzc we'll need `AST_Super` for `harmony`

(Fun fact: `AST_This.name` is always equal to `"this"`)

Reply:
Might be splitting hairs here, but shouldn't `AST_SymbolRef` be first? Aren't `AST_SymbolRef`s more common than `AST_PropAccess`? Could be a toss up.

I wonder if there'd be any merit to having `AST_Super` be derived from `AST_This` to simplify the code. They appear to have the same characteristics and issues time and time again. And the cases in `lib/compress.js` that diverge with respect to `AST_This` versus `AST_Super` logic - `has_overlapping_symbol`, `OPT(AST_Call)` and `contains_this` - is that by design?

> AST_This.name is always equal to "this"

Is that an ES thing or an uglify thing?

Reply:
FWIW, I just verified that if `AST_Super` is derived from `AST_This` and all references to `AST_Super` are removed from `lib/compress.js`, then the test suite still runs successfully. I don't know if this is a consequence of incomplete `super` test coverage or whether it's correct, but it is something to consider.

Reply:
> Might be splitting hairs here, but shouldn't `AST_SymbolRef` be first? Aren't `AST_SymbolRef`s more common than `AST_PropAccess`? Could be a toss up.

I was just doing an alphabetical sort, but may be there is some room for optimisation there.

> FWIW, I just verified that if `AST_Super` is derived from `AST_This`...

Good to know - I'll have a look now and cook a PR up if it works out. There is precedence - `AST_ForOf` is derived from `AST_ForIn`, for instance.